server/src/unifyfs_cmd_handler.c

```diff
@@ -406,23 +398,17 @@ static void unifyfs_metaset_rpc(hg_handle_t handle)
     assert(hret == HG_SUCCESS);
 
     /* store file name for given global file id */
-    unifyfs_file_attr_t fattr;
-    memset(&fattr, 0, sizeof(fattr));
     int create         = (int) in.create;
-    fattr.gfid         = in.gfid;
-    strncpy(fattr.filename, in.filename, sizeof(fattr.filename));
-    fattr.mode         = in.mode;
-    fattr.uid          = in.uid;
-    fattr.gid          = in.gid;
-    fattr.size         = in.size;
-    fattr.atime        = in.atime;
-    fattr.mtime        = in.mtime;
-    fattr.ctime        = in.ctime;
-    fattr.is_laminated = in.is_laminated;
+
+    unifyfs_file_attr_t fattr;
+    fattr = in.attr;
 
     /* if we're creating the file,
      * we initialize both the size and laminate flags */
-    int ret = unifyfs_set_file_attribute(create, create, &fattr);
+    //int ret = unifyfs_set_file_attribute(create, create, &fattr);
```

Comment:
There is some subtle but important logic in the old unifyfs_set_file_attribute() implementation.  In particular, the first two parameters passed to that function are boolean flags that indicate whether the existing size and is_laminated fields should be preserved or replaced.  You can see what I mean if you look at the function code here:

https://github.com/LLNL/UnifyFS/blob/ec593f35f36a59ed5d3aeb84dbf053f50674172e/server/src/unifyfs_metadata.c#L212

I think we'll still want to keep that logic in the new implementation.

Reply:
I uncommented this function/line for now. Probably, we want to put the logic somewhere else if we are to get rid of mdhim.